src/Rules/Deprecations/TypeHintDeprecatedInFunctionSignatureRule.php

```diff
@@ -0,0 +1,125 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\Deprecations;
+
+use PhpParser\Node;
+use PHPStan\Analyser\Scope;
+use PHPStan\Broker\Broker;
+use PHPStan\Node\InFunctionNode;
+use PHPStan\Reflection\ClassReflection;
+use PHPStan\Reflection\FunctionReflection;
+use PHPStan\Reflection\ParameterReflection;
+use PHPStan\Reflection\ParametersAcceptorSelector;
+
+class TypeHintDeprecatedInFunctionSignatureRule implements \PHPStan\Rules\Rule
+{
+
+	/** @var Broker */
+	private $broker;
+
+	public function __construct(Broker $broker)
+	{
+		$this->broker = $broker;
+	}
+
+	public function getNodeType(): string
+	{
+		return InFunctionNode::class;
+	}
+
+	/**
+	 * @param InFunctionNode $node
+	 * @param \PHPStan\Analyser\Scope $scope
+	 * @return string[] errors
+	 */
+	public function processNode(Node $node, Scope $scope): array
+	{
+		if (DeprecatedScopeHelper::isScopeDeprecated($scope)) {
+			return [];
+		}
+
+		/** @var FunctionReflection $function */
+		$function = $scope->getFunction();
+		if (!$function instanceof FunctionReflection) {
+			throw new \PHPStan\ShouldNotHappenException();
+		}
+		$functionSignature = ParametersAcceptorSelector::selectSingle($function->getVariants());
+
+		$errors = [];
+		foreach ($functionSignature->getParameters() as $i => $parameter) {
+			/** @var ParameterReflection $parameter */
```

Comment:
I don't think this `@var` has to exist - `$functionSignature->getParameters()` is already properly annotated.